interpreter/simplecompositevalue.go

```diff
@@ -43,6 +43,11 @@ type SimpleCompositeValue struct {
 	// This is used for distinguishing between transaction values and other composite values.
 	// TODO: maybe cleanup if there is an alternative/better way.
 	isTransaction bool
+
+	// Metadata is a property bag to carry internal data
+	// that are not visible to cadence users.
+	// TODO: any better way to pass down information?
+	metadata map[string]Value
```

Comment:
Maybe we can name this "private fields"?

Reply:
Renamed https://github.com/onflow/cadence/pull/3830/commits/7f6a2f40be22bfe147003c76d18999615e506049